simulation_bridge/src/utils/performance_monitor.py

```diff
@@ -0,0 +1,282 @@
+    output_dir: Path = Path("performance_logs")
+    csv_path: Path = field(
+        default_factory=lambda: Path("performance_logs/performance_metrics.csv"))
+
+
+@dataclass
+class _RuntimeState:
+    """Internal runtime state for the monitor."""
+    metrics_by_operation_id: Dict[str, PerformanceMetrics] = field(
+        default_factory=dict)
+    metrics_history: List[PerformanceMetrics] = field(default_factory=list)
+    process: Optional[psutil.Process] = None
+
+
+class EventType(Enum):
+    """Supported performance events."""
+    CORE_RECEIVED_INPUT = "core_received_input"
+    CORE_SENT_INPUT = "core_sent_input"
+    CORE_RECEIVED_RESULT = "core_received_result"
+    RESULT_SENT = "result_sent"
+
+
+class PerformanceMonitor:
+    _instance: Optional[PerformanceMonitor] = None
+    _initialized: bool = False
+
+    def __new__(cls, *args, **kwargs):
+        if cls._instance is None:
+            cls._instance = super().__new__(cls)
+        return cls._instance
+
+    def __init__(self, config: Optional[Dict[str, Any]] = None):
+        if self._initialized:
+            return
+
+        self._enabled: bool = False
+        self.paths = _Paths()
+        self._state = _RuntimeState()
+
+        if config:
+            perf_cfg = config.get("performance", {})
+            self._enabled = perf_cfg.get("enabled", False)
+            log_file = perf_cfg.get("file", str(self.paths.csv_path))
+            self.paths.output_dir = Path(log_file).parent
+            self.paths.csv_path = Path(log_file)
+
+        if self._enabled:
+            try:
+                self.paths.output_dir.mkdir(parents=True, exist_ok=True)
+                self._state.process = psutil.Process()
+                if not self.paths.csv_path.exists():
+                    self._write_csv_headers()
+                logger.debug("PERFORMANCE - Logging to %s", self.paths.csv_path)
+            except Exception as exc:
+                logger.error(
+                    "Failed to initialize performance monitoring: %s", exc)
+                self._enabled = False
+        else:
+            logger.debug("PERFORMANCE - Monitoring disabled")
+
+        self._initialized = True
+
+    @property
+    def enabled(self) -> bool:
+        """Return whether performance monitoring is active."""
+        return self._enabled
+
+    @property
+    def history(self) -> List[PerformanceMetrics]:
+        """Return the list of completed metrics."""
+        return list(self._state.metrics_history)
+
+    def get_metric(self, operation_id: str) -> Optional[PerformanceMetrics]:
+        """Return metrics for an ongoing operation."""
+        return self._state.metrics_by_operation_id.get(operation_id)
+
+    def _write_csv_headers(self) -> None:
+        if not self._enabled:
+            return
+        try:
+            with self.paths.csv_path.open("w", newline="", encoding="utf-8") as fh:
+                writer = csv.writer(fh)
+                writer.writerow([
```

Comment:
Since these fields are essentially attributes of the PerformanceMetrics class (which will hopefully split into smaller classes), can these column headers be generated from class attributes?